worker/run.py

```diff
@@ -28,8 +30,17 @@ async def main():
         task_types=[TaskType.TRANSCRIBE],
     )
     while True:
-        no_work = await worker.run_task()
-        if no_work:
+        try:
+            no_work = await worker.run_task()
+            if no_work:
+                await asyncio.sleep(5)
+        except requests.exceptions.ConnectionError:
+            logging.warn("could not connect to backend")
+            await asyncio.sleep(5)
+        except Exception:
```

Comment:
Are we sure in this case the `worker` is still valid?
Maybe we want to recreate the worker completely in this case?

Reply:
I also thought about that (also validating the connection in the constructor) but I didn't see any immediate advantage and think it is fine like that for now

Reply:
Let's cross that bridge once we get there. Currently we also have a blanket except in `Worker.run_task` which will capture any exceptions from the actual task code, which I think would be the main source of state corruption, so this needs more work